src/Form/CoyoteImgBatchForm.php

```diff
@@ -0,0 +1,154 @@
+    if (self::isDefined($token) && self::isDefined($endpoint)) {
+      $profile = CoyoteApiClientHelperFunctions::getProfile($endpoint, $token);
+      $role = self::getProfileRole($profile);
+    }
+     
+    $disabled = false;
+    if (!self::isDefined($token) || !self::isDefined($endpoint) || !self::isDefined($organizationId) || !self::isDefined($resourceGroup) || is_null($profile) || is_null($role)) {
+         $disabled=true;
+    }
+ 
+    $form['coyote_message'] = [
+       '#type' => 'item',
+        '#markup' => "<strong>".$this->t('Values used'). "</strong><br />".$this->t('Coyote API token:')." ".$token."<br />". $this->t('Coyote API endpoint:') ." ".$endpoint. "<br />" . $this->t('Organization ID:'). " ".$organizationId . "<br />". $this->t('Resource Group:'). " ".$resourceGroup."<br />".$this->t("Permissions:")." ".$role,
+    ];
+
+    if ($disabled) {
+       $form['coyote_warning'] = [
+          '#type' => 'item',
+          '#markup' => "<strong>".  $this->t("Invalid data, correct in Settings")."</strong>",
+       ];
+          
+    }
+
+    $form['batch_processing'] = [
+      '#type' => 'submit',
+      '#disabled' => $disabled,
+      '#value' => $this->t('Batch processing of all images?'),
+    ];
+
+    return $form;
+  }
+
+  public function submitForm(array &$form, FormStateInterface $form_state): void {
+     $nids = \Drupal::entityQuery('node')->execute();
+
+     foreach ($nids as $nid){     
+       $operations[] = [['\Drupal\coyote_img_desc\Form\CoyoteImgBatchForm','batch_processing_nodes'], [$nid]];
+     }
+     $batch = [
+          'title' => $this->t('Batch processing All Nodes ...'),
+          'operations' => $operations,
+        //  'finished' => ['Drupal\coyote_img_desc\Form\CoyoteImgBatchForm','batch_processing_nodes_finished'],
+      ];
+      batch_set($batch);
+  }
+  
+  public function batch_processing_nodes($nid, &$context) {
+     $message = 'Proccessing ALL nodes to coyote';
+
+     $results = array();
+   
+     $config = \Drupal::config('coyote_img_desc.settings');
+
+     $entity_type = 'node';
+     $view_mode = 'default';
+
+     $viewBuilder = \Drupal::entityTypeManager()->getViewBuilder($entity_type);
+     $storage = \Drupal::entityTypeManager()->getStorage($entity_type);
+     $node = $storage->load($nid);
+     $build = $viewBuilder->view($node, $view_mode);
+     $build['_coyote_node_url'] = $node->toUrl('canonical', ['absolute' => true])->toString();
+     $hostUri = $node->toUrl('canonical', ['absolute' => true])->toString();
+     $output = render($build);
+
+     $isPublished = $node->isPublished();
+     $coyoteProcessUnpublishedNodes = $config->get('coyote_process_unpublished_nodes');
+
+     if (!$coyoteProcessUnpublishedNodes && !$isPublished) return false;
+     $contentHelper = new ContentHelper($output);
+
+     $images = $contentHelper->getImages();
+     foreach($images as $image) {
+        $resource = Util::getImageResource($image, $hostUri);
```

Comment:
`$resource` isn't used anywhere, we probably want to track (log?) if this is `null`.

Reply:
one extra commit. Please have a look.